packages/fxa-shared/subscriptions/metadata.ts

```diff
@@ -169,3 +172,152 @@ export const getProductSupportApps =
       return acc;
     }, {});
   };
+
+export const getPlanProductConfig = (plan: AbbrevPlan) => {
+  const configuration = plan.configuration;
+
+  if (!configuration) {
+    throw new Error('Plan configuration not found');
+  }
+
+  return configuration;
+};
+
+export const flattenProductConfig = (
+  plan: AbbrevPlan,
+  userLocales: readonly string[] = [DEFAULT_LOCALE]
+) => {
+  let output: {
+    uiContent: ProductConfig['uiContent'];
+    urls: ProductConfig['urls'];
+    support: ProductConfig['support'];
+  };
+
+  const selectedLocale =
+    parser.pick(availableLocales, userLocales.join(',')) || DEFAULT_LOCALE;
+
```

Comment:
The available locales are not the pre-defined list in this case.  It's the default ('en') and whatever's in the config object's `locales` object, that is, the keys of plan.configuration.locales.

Reply:
Ah OK nice, that makes sense.

I made some changes, but something doesn't feel right, and I get a sense its something obvious. I'd appreciate another look if you can please.